### DISCUSSION TITLE
About how to fill the surface data of several pyvista.PolyData internally?

### DISCUSSION THREAD

**Discussion by Finn0395** [pyvista/pyvista#4741] [Q&A]
I have several PolyData data of three grids. These three data enclose the surface of a volume. I want to fill the interior of this volume and get an irregular volume with a non-empty interior. Can you help me? thank you very much

**Comment by tkoyama010** [comment]
Maybe tetgen is your solution.
![sphx_glr_cow_003](https://github.com/pyvista/pyvista/assets/7513610/3302e57b-a3e6-4369-a61f-d7e2c4145a5c)
https://tetgen.pyvista.org/examples/cow.html#sphx-glr-examples-cow-py
If you share your PolyData with us, we can comment in more detail.
